core/src/test/scala/org/apache/gearpump/cluster/MasterHarness.scala

```diff
@@ -130,13 +138,24 @@ trait MasterHarness {
     mainObj.getClass.getName.dropRight(1)
   }
 
+  import Constants._
+
   def getMasterListOption(): Array[String] = {
-    Array(s"-D${Constants.GEARPUMP_CLUSTER_MASTERS}.0=$getHost:$getPort",
-    s"-D${Constants.GEARPUMP_HOSTNAME}=$getHost")
+    masterProperties.asScala.toList.map { kv =>
```

Comment:
This may break passing a list as property via -D and needs to be tested and verified. Note the .0 which was required as a workaround.